ui/components/failed.jsx

```diff
@@ -0,0 +1,175 @@
+import React from 'react';
+import ImageDiff from 'react-image-diff';
+import {Clearfix, ButtonGroup, Button,
+  Jumbotron, Grid, Row, Col, Panel}
+  from 'react-bootstrap';
+
+function _renderDefaultView(paths) {
+  return <div className="simple">
+    <img className="diff"
+         src={paths.diff}
+         key={paths.diff} />
+  </div>;
+}
+
+function _render2UpView(paths) {
+  return <div className="simple">
+    <Grid>
+      <Row className="show-grid">
+        <Col xs={12} md={6}>
+          <img className="baseline"
+               src={paths.baseline}
+               key={paths.baseline} />
+        </Col>
+        <Col xs={12} md={6}>
+          <img className="screenshot"
+               src={paths.screenshot}
+               key={paths.screenshot} />
+        </Col>
+      </Row>
+    </Grid>
+  </div>;
+}
+
+function _renderSwipeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="swipe"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _renderFadeView(paths, value, onValueChange) {
+  return <div className="special">
+    <ImageDiff before={paths.screenshot}
+               after={paths.baseline}
+               type="fade"
+               value={value} />
+    <Clearfix />
+    <input type="range"
+           min={0}
+           max={1}
+           step={.01}
+           defaultValue={value}
+           onChange={onValueChange} />
+  </div>;
+}
+
+function _getSelectViewButtons(viewOptions, currentView, onViewChange) {
+  let buttons = [];
+
+  viewOptions.forEach(function(item) {
+    buttons.push(
+      <Button name={item}
+              key={item}
+              onClick={onViewChange}
+              className={item === currentView ? 'active' : null}>
```

Comment:
A nice way to treat classes like this is to use [classnames](https://github.com/JedWatson/classnames). Random web-ui [example](https://github.hootops.com/analytics/web-ui/blob/aa3bc513bbc2d5497daae08d398a15a7cd0feb0e/src/scripts/components/board-widgets.jsx#L275)